lib/Sema/CSApply.cpp

```diff
@@ -7577,7 +7577,11 @@ Expr *ExprRewriter::finishApply(ApplyExpr *apply, Type openedType,
   // FIXME: handle unwrapping everywhere else
   assert(!unwrapResult);
 
-  assert(!cs.getType(fn)->is<UnresolvedType>());
+  // If this is an UnresolvedType in the system, preserve it.
+  if (cs.getType(fn)->is<UnresolvedType>()) {
```

Comment:
@slavapestov It looks like some of the recently removed code paths are still used by the code completion. Maybe we should revert https://github.com/apple/swift/pull/36028?

Reply:
Is code completion still submitting UnresolvedType ASTs for type checking?

Reply:
I assume so 🤷

Reply:
@xedin could we do this with holes?

Reply:
Not all of the code completion requests has been ported to use `typeCheckForCodeCompletion` yet, but the plan is to avoid having to apply solution and use holes everywhere. Meanwhile I think the most sensible thing would be to revert #36028.